File: Java/TimeConversion.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;

public class TimeConversion {

    /*
     * Complete the timeConversion function below.
     */
    static String timeConversion(String s) {
        /*
         * Write your code here.
         */
        if (s.substring(0, 2).equals("12"))
        {
            String[] array = s.split(":");
            int newTime = Integer.parseInt(array[0]) - 12;
            if (newTime == 0)
                s = "00" + s.substring(2);
            else
                s = newTime + s.substring(2);
            System.out.println(s);
        }
        if (s.substring(s.length() - 2).equals("AM"))
            return s.substring(0, s.length() - 2);
        else
        {
            String[] array = s.split(":");
            int newTime = Integer.parseInt(array[0]) + 12;
            return newTime + s.substring(2, s.length() - 2);
        }
    }

    private static final Scanner scan = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = scan.nextLine();

        String result = timeConversion(s);

        bw.write(result);
        bw.newLine();

        bw.close();
    }
}

File: TimeConversion.py
def timeConversion(s):
    arr = s.split(":")
    arr.append(arr[-1][2:])
    arr[-2] = arr[-2][:2]
    if arr[-1] == "PM":
        if arr[0] != "12":
            new_hour = int(arr[0]) + 12
            if new_hour == 24:
                new_hour = "00"
            arr[0] = str(new_hour)
    else:
        if arr[0] == "12":
            arr[0] = "00"
    return arr[0] + ":" + arr[1] + ":" + arr[2]


if __name__ == '__main__':
    print(timeConversion("12:01:01PM"))